includes/class-give-cron.php

```diff
@@ -99,6 +138,98 @@ private function daily_events() {
 		}
 	}
 
+	/**
+	 * Schedule asyncly events
+	 *
+	 * @since  1.8.13
+	 * @access private
+	 *
+	 * @return void
+	 */
+	private function asyncly_events() {
+		if ( ! wp_next_scheduled( 'give_asyncly_scheduled_events' ) ) {
+			wp_schedule_event( current_time( 'timestamp' ), 'asyncly', 'give_asyncly_scheduled_events' );
+		}
+	}
+
+	/**
+	 * get cron job action name
+	 *
+	 * @since  1.8.13
+	 * @access public
+	 *
+	 * @param string $type
+	 *
+	 * @return string
+	 */
+	public static function get_cron_action( $type = 'weekly' ) {
+		switch ( $type ) {
+			case 'daily':
+				$cron_action = 'give_daily_scheduled_events';
+				break;
+
+			case 'asyncly':
+				$cron_action = 'give_asyncly_scheduled_events';
+				break;
+
+			default:
+				$cron_action = 'give_weekly_scheduled_events';
+				break;
+		}
+
+		return $cron_action;
+	}
+
+	/**
+	 * Add action to cron action
+	 *
+	 * @since  1.8.13
+	 * @access private
+	 *
+	 * @param        $action
+	 * @param string $type
+	 */
+	private static function add_event( $action, $type = 'weekly' ) {
+		$cron_event = self::get_cron_action( $type );
+		add_action( $cron_event, $action );
+	}
+
+	/**
+	 * Add weekly event
+	 *
+	 * @since  1.8.13
+	 * @access public
+	 *
+	 * @param $action
+	 */
+	public static function add_weekly_event( $action ) {
+		self::add_event( $action, 'weekly' );
+	}
+
+	/**
+	 * Add daily event
+	 *
+	 * @since  1.8.13
+	 * @access public
+	 *
+	 * @param $action
+	 */
+	public static function add_daily_event( $action ) {
+		self::add_event( $action, 'daily' );
+	}
+
+	/**
+	 * Add asyncly event
+	 *
+	 * @since  1.8.13
+	 * @access public
+	 *
+	 * @param $action
+	 */
+	public static function add_asyncly_event( $action ) {
```

Comment:
@DevinWalker `asyncly` cron job which will help us to process anything in background for example if we want to sync `form id` in log when change for `form id` of payment then this process can process in background instead of instant